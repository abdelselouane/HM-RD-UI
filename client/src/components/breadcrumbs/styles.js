import styled from 'styled-components';

export const Container = styled.div`
  grid-area: cmess;

  .breadcrumb {
    font-size: 16px;
  }
`;

export default Container;
