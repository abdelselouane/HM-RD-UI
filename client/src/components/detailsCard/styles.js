import styled from 'styled-components';

export const Card = styled.div`
  width: 85%;
  margin-bottom: 20px;

  .card-content {
    padding: 0px 25px;
  }
`;

export const Title = styled.span`
  align-items: flex-end;
  display: flex;
  justify-content: flex-start;
  width: 100%;
  border-bottom: solid #666 1px;
  padding-bottom: 15px;

  h2 {
    display: inline-block;
    color: #f96302;
    font-weight: 900;
    padding-right: 25px;
  }

  span {
    text-transform: capitalize;
  }
`;

export const Row = styled.div`
  display: flex;
  padding: 15px;
`;

export const Col = styled.div`
  ${props => props.flex && `flex: ${props.flex}`}
  display: flex;
  flex-direction: column;
`;

export const Label = styled.div`
  font-weight: 700;
  color: #666;
`;
