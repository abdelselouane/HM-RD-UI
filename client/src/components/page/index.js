export { default } from './page';
export { Container } from './styles';
