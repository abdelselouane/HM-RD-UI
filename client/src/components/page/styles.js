import styled, { css } from 'styled-components';

const defaultGrid = css`
  display: grid;
  grid-template-rows: 70px
                      48px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 72px);
  grid-template-columns: 20px 1fr repeat(4, 3fr) 1fr 20px;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . cmess cmess cmess cmess . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . carea carea carea carea . ."
                        ". . . . . . . .";
`;

const lowResolutionGrid = css`
  display: grid;
  grid-template-areas:  ". . . . . . . ."
                        ". . . . . . . ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". cmess cmess cmess cmess cmess cmess ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". carea carea carea carea carea carea ."
                        ". . . . . . . .";
  grid-template-rows: 70px
                      24px
                      24px
                      24px
                      repeat(3, 1fr)
                      repeat(1, 48px);
  grid-template-columns: 10px 1fr repeat(4, 3fr) 1fr 10px;
`;

export const GridLayout = styled.div`
  background-color: ${({ theme }) => theme.black12};
  box-sizing: border-box;
  color: ${({ theme }) => theme.blackA11};
  height: 100%;  
  ${props => (props.enableScroll ? 'overflow-x: scroll;' : 'overflow: hidden;')}
  
  ${() => defaultGrid}
  @media only screen and (max-width: 1024px) {
    ${() => lowResolutionGrid}
  }
`;

export const Container = styled.div`
  align-items: flex-start;
  display: flex;
  grid-area: carea;
  justify-content: center;
  ${({ hideBoxShadow }) => (hideBoxShadow ? '' : 'box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);')}
`;
