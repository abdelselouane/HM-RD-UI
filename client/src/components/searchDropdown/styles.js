import styled from 'styled-components';

const StyledSearchDropdown = styled.div`
  display: block;
  width: 700px;
  position: absolute;
  top: 70%;
  box-shadow: 0 0 2px 0 rgba(0, 0, 0, 0.12), 0 2px 2px 0 rgba(0, 0, 0, 0.24);
  z-index: 5;

  .scrollable-table {
    overflow-y: hidden;
  }

  .standard-table tbody tr {
    td {
      background-color: ${({ theme }) => theme.white} !important;
    }

    &:hover td {
      background-color: ${({ theme }) => theme.black10} !important;
    }
  }

  .table-body.alternating-rows {
    width: 100%;
    
    tr:nth-child(2n + 1),
    tr:nth-child(2n) {
      background-color: ${({ theme }) => theme.black10};
    }

    tr:hover td,
    tr:focus td {
      background-color: ${({ theme }) => theme.black12} !important;
    }
    
    tr td {
      border: none;
    }
  }
`;

export default StyledSearchDropdown;
