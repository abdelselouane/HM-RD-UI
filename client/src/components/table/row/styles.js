import styled from 'styled-components';

const StyledTd = styled.td`
  ${props => {
    if (props.tdProps
      && props.tdProps.color
      && Object.keys(props.theme).includes(props.tdProps.color)) {
      return {
        ...props.tdProps,
        color: `${props.theme[props.tdProps.color]} !important`
      };
    }
    if (props.disableRow) {
      return {
        ...props.tdProps,
        color: `${props.theme.black14} !important`
      };
    }
    return props.tdProps;
  }}
`;

export default StyledTd;
