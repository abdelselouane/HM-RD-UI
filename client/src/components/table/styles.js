import styled, { css } from 'styled-components';

const expandableTableCss = css`
  &.standard-table > thead > tr > th {
    background-color: ${({ theme }) => theme.white} !important; 
    &.sortable {
      &:after {
        position: static !important;
        padding-left: 10px;
      }
    }  
  }

  &.standard-table > tbody > tr {
    > td {
      border-bottom: 1px solid ${({ theme }) => theme.black14};
    }

    &:nth-child(4n + 1):hover > td,
    &:nth-child(4n + 2):hover > td,
    &:nth-child(4n + 3):hover > td,
    &:nth-child(4n + 4):hover > td {
      background-color: unset !important;
    }

    &:nth-child(4n + 1) td,
    &:nth-child(4n + 2) td {
      background-color: ${({ theme }) => theme.black11} !important;
    }

    &:nth-child(4n + 3) td,
    &:nth-child(4n + 4) td {
      background-color: ${({ theme }) => theme.white} !important;
    }

    &[id^='expandable'] {
      &:nth-child(4n + 1):hover,
      &:nth-child(4n + 1):hover > td,
      &:nth-child(4n + 2):hover,
      &:nth-child(4n + 2):hover > td {
        background-color: ${({ theme }) => theme.black11} !important;
      }

      &:nth-child(4n + 3):hover,
      &:nth-child(4n + 3):hover > td,
      &:nth-child(4n + 4):hover,
      &:nth-child(4n + 4):hover > td {
        background-color: ${({ theme }) => theme.white} !important;
      }
    }
  }
`;

const clearOnHoverCss = css`
  &.standard-table > thead > tr:hover,
  &.standard-table > thead > tr:hover > th {
    background-color: inherit !important;
  }
`;

export const StyledTable = styled.table`
  background-color: unset !important;
  ${({ isExpandable }) => isExpandable && expandableTableCss} ${({ clearOnHover }) =>
  clearOnHover && clearOnHoverCss};
`;

export const StyledTh = styled.th`
  ${props => {
    if (
      props.thProps &&
      props.thProps.color &&
      Object.keys(props.theme).includes(props.thProps.color)
    ) {
      return {
        ...props.thProps,
        color: `${props.theme[props.thProps.color]} !important`
      };
    }
    return props.thProps;
  }};
`;
