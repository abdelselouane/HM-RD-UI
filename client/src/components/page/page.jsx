/* global sessionStorage */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import { GridLayout } from './styles';

import Header from '../header';
import Notification from '../notification';
import Breadcrumbs from '../breadcrumbs';

import { withContext } from '../../contexts/history';

import * as LoginActions from '../../actions/loginActions';
import * as NotifyActions from '../../actions/notificationActions';
import { loadConfig } from '../../actions/configActions';
import { clearResults } from '../../actions/partsSearchActions';
import { clearCart } from '../../actions/cartActions';

export class Page extends Component {
  static propTypes = {
    actions: PropTypes.shape({
      clearCart: PropTypes.func.isRequired,
      clearResults: PropTypes.func.isRequired,
      logoutUser: PropTypes.func.isRequired,
      hideMessage: PropTypes.func.isRequired,
      loadConfig: PropTypes.func.isRequired
    }).isRequired,
    enableScroll: PropTypes.bool,
    children: PropTypes.node.isRequired,
    className: PropTypes.string.isRequired,
    empty: PropTypes.bool,
    notify: PropTypes.shape(),
    onCloseNotification: PropTypes.func,
    showHomeNav: PropTypes.bool,
    showReturn: PropTypes.bool,
    userProfile: PropTypes.shape(),
    router: PropTypes.shape({
      location: PropTypes.shape({
        pathname: PropTypes.string.isRequired
      })
    }).isRequired,
    history: PropTypes.shape({
      goBack: PropTypes.func.isRequired,
      push: PropTypes.func.isRequired
    }).isRequired,
    showBreadcrumb: PropTypes.bool
  };

  static defaultProps = {
    empty: false,
    enableScroll: false,
    notify: null,
    onCloseNotification: null,
    showBreadcrumb: true,
    showHomeNav: false,
    showReturn: false,
    userProfile: null
  };

  componentWillMount() {
    this.clearData();
    this.props.actions.loadConfig();
  }

  clearData = () => {
    const { pathname: nextLocation } = this.props.router.location;
    const clearRoutes = ['/', '/home'];
    if (clearRoutes.includes(nextLocation)) {
      this.props.actions.clearResults();
      this.props.actions.clearCart();
      this.props.actions.hideMessage();
    }
  };

  handleLogout = () => {
    this.props.actions.logoutUser();
    const userSession = sessionStorage.getItem('userProfile');
    if (!userSession && this.props.router.location.pathname !== '/') {
      this.props.history.push('/');
      this.props.actions.hideMessage();
    }
  };

  handleCloseNotification = () => {
    if (this.props.onCloseNotification) {
      this.props.onCloseNotification();
    }
    this.props.actions.hideMessage();
  };

  handleLogoClick = () => {
    this.props.history.push('/home');
  }


  handleAdminClick = () => {
    this.props.history.push('/admin');
  }

  render() {
    const { className, notify, enableScroll } = this.props;
    return (
      <GridLayout
        className={`${className}-page`}
        enableScroll={enableScroll}
      >
        <Header
          showActions={!this.props.empty}
          userProfile={this.props.userProfile}
          handleLogoutClick={this.handleLogout}
          handleLogoClick={this.handleLogoClick}
          handleAdminClick={this.handleAdminClick}
        />
        <Notification
          {...notify}
          onClick={() => this.handleCloseNotification()}
          isOpen={notify.title.length > 0}
        />
        { this.props.showBreadcrumb && <Breadcrumbs /> }
        {this.props.children}
      </GridLayout>
    );
  }
}

const mapStateToProps = state => ({
  userProfile: state.login.userProfile,
  router: state.router,
  notify: state.notify
});

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators(
    { ...NotifyActions, ...LoginActions, clearCart, clearResults, loadConfig },
    dispatch
  )
});

const PageWithContext = withContext(Page);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(PageWithContext);
