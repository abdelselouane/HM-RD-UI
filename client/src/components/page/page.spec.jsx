/* global beforeAll, describe, expect, jest, it */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Page } from './page';
import Header from '../header';
import Notification from '../notification';
import * as Styles from './styles';
import { isRegExp } from 'util';

describe('<Page />', () => {
  let wrapper;
  let wrapperInstance;
  let pageProps;

  beforeAll(() => {
    global.sessionStorage.removeItem('userProfile');

    pageProps = {
      className: 'test',
      onCloseNotification: jest.fn(),
      empty: false,
      userProfile: { id: 'mockuser' },
      notify: {
        title: '',
        details: '',
        className: 'mockNotifyClass',
        icon: 'mockIcon'
      },
      actions: {
        logoutUser: jest.fn(),
        clearResults: jest.fn(),
        clearCart: jest.fn(),
        hideMessage: jest.fn(),
        loadConfig: jest.fn()
      },
      history: { goBack: jest.fn(), push: jest.fn() },
      router: { location: { pathname: '/' } }
    };

    wrapper = shallow(
      <Page {...pageProps}>
        <div />
      </Page>
    );

    wrapperInstance = wrapper.instance();
  });

  it('should render properly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render styles properly', () => {
    Object.values(Styles).forEach(Component => {
      expect(toJson(shallow(<Component />))).toMatchSnapshot();
    });

    expect(toJson(shallow(<Styles.Container hideBoxShadow />))).toMatchSnapshot();
  });

  it('should call logoutUser action and push to login', () => {
    wrapper.setProps({
      ...pageProps,
      history: { ...pageProps.history },
      router: { location: { pathname: '/home' } }
    });

    const spy = jest.spyOn(pageProps.actions, 'logoutUser');
    const spy2 = jest.spyOn(pageProps.history, 'push');

    wrapperInstance.handleLogout();

    expect(spy).toHaveBeenCalled();
    expect(spy2).toHaveBeenCalledWith('/');

    spy.mockReset();
    spy.mockRestore();
    spy2.mockReset();
    spy2.mockRestore();
  });

  describe('Notification', () => {
    it('should render Notification with isOpen true if notify.title has length > 0', () => {
      wrapper.setProps({ notify: { title: 'mockTitle' } });
      expect(wrapper.find(Notification).prop('isOpen')).toBe(true);
    });

    it('should render Notification with isOpen false if notify.title has length > 0', () => {
      wrapper.setProps({ notify: { title: '' } });
      expect(wrapper.find(Notification).prop('isOpen')).toBe(false);
    });

    it('should call closeNotification and hideMessage in Notification onClick function', () => {
      const spy = jest.spyOn(pageProps, 'onCloseNotification');
      const spy2 = jest.spyOn(pageProps.actions, 'hideMessage');
      wrapper.setProps({ notify: { title: 'mockTitle' } });
      wrapper.find(Notification).shallow().find('.close').simulate('click');

      expect(spy).toHaveBeenCalled();
      expect(spy2).toHaveBeenCalled();
      spy.mockReset();
      spy.mockRestore();
      spy2.mockReset();
      spy2.mockRestore();
    });
  });

  describe('logoutButton', () => {
    it('should not push "/" to history if handleLogout is called while userSession is defined and pathName is "/"', () => {
      global.sessionStorage.setItem('userProfile', {});
      const spy = jest.spyOn(pageProps.history, 'push');
      wrapper.find(Header).shallow().find('.logout-btn').simulate('click');
      expect(spy).not.toHaveBeenCalled();
      spy.mockReset();
      spy.mockRestore();
    });
  });

  describe('componentWillMount', () => {
    it('should call clearResults and clearCart actions if current location is "/"', () => {
      const spy = jest.spyOn(pageProps.actions, 'clearResults');
      const spy2 = jest.spyOn(pageProps.actions, 'clearCart');
      spy.mockReset();
      spy2.mockReset();

      wrapper.setProps({
        router: { location: { pathname: '/' } }
      });
      wrapper.instance().componentWillMount();

      expect(spy).toHaveBeenCalled();
      expect(spy2).toHaveBeenCalled();

      spy.mockReset();
      spy.mockRestore();
      spy2.mockReset();
      spy2.mockRestore();
    });

    it('should call loadConfig actions when page loads', () => {
      const spy = jest.spyOn(pageProps.actions, 'loadConfig');
      spy.mockReset();

      wrapper.instance().componentWillMount();

      expect(spy).toHaveBeenCalled();
      spy.mockReset();
      spy.mockRestore();
    });

    it('should call clearResults and clearCart actions if current location is "/home"', () => {
      const spy = jest.spyOn(pageProps.actions, 'clearResults');
      const spy2 = jest.spyOn(pageProps.actions, 'clearCart');
      spy.mockReset();
      spy2.mockReset();

      wrapper.setProps({
        router: { location: { pathname: '/home' } }
      });
      wrapper.instance().componentWillMount();

      expect(spy).toHaveBeenCalled();
      expect(spy2).toHaveBeenCalled();

      spy.mockReset();
      spy.mockRestore();
      spy2.mockReset();
      spy2.mockRestore();
    });

    it('should not call clearResults and clearCart actions if current location is not "/" or "/home"', () => {
      const spy = jest.spyOn(pageProps.actions, 'clearResults');
      const spy2 = jest.spyOn(pageProps.actions, 'clearCart');
      spy.mockReset();
      spy2.mockReset();

      wrapper.setProps({
        history: { ...pageProps.history },
        router: { location: { pathname: '/orderParts' } }
      });
      wrapper.instance().componentWillMount();

      expect(spy).not.toHaveBeenCalled();
      expect(spy2).not.toHaveBeenCalled();

      spy.mockReset();
      spy.mockRestore();
      spy2.mockReset();
      spy2.mockRestore();
    });
  });

  describe('handleLogoClick', () => {
    it('should push home to history on handleLogoClick call', () => {
      const spy = jest.spyOn(pageProps.history, 'push');
      wrapper.instance().handleLogoClick();
      expect(spy).toHaveBeenCalledWith('/home');
      spy.mockReset();
    });
  });
});
