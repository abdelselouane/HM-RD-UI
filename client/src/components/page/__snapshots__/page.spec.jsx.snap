// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> should render properly 1`] = `
<styled.div
  className="test-page"
  enableScroll={false}
>
  <Header
    handleAdminClick={[Function]}
    handleLogoClick={[Function]}
    handleLogoutClick={[Function]}
    showActions={true}
    userProfile={
      Object {
        "id": "mockuser",
      }
    }
  />
  <Notification
    className="mockNotifyClass"
    details=""
    icon="mockIcon"
    isOpen={false}
    onClick={[Function]}
    title=""
  />
  <Connect(Breadcrumbs) />
  <div />
</styled.div>
`;

exports[`<Page /> should render styles properly 1`] = `
.c0 {
  box-sizing: border-box;
  height: 100%;
  overflow: hidden;
  display: grid;
  grid-template-rows: 70px 48px 24px 24px repeat(3,1fr) repeat(1,72px);
  grid-template-columns: 20px 1fr repeat(4,3fr) 1fr 20px;
  grid-template-areas: ". . . . . . . ." ". . . . . . . ." ". . cmess cmess cmess cmess . ." ". . cmess cmess cmess cmess . ." ". . carea carea carea carea . ." ". . carea carea carea carea . ." ". . carea carea carea carea . ." ". . . . . . . .";
}

@media only screen and (max-width:1024px) {
  .c0 {
    display: grid;
    grid-template-areas: ". . . . . . . ." ". . . . . . . ." ". cmess cmess cmess cmess cmess cmess ." ". cmess cmess cmess cmess cmess cmess ." ". carea carea carea carea carea carea ." ". carea carea carea carea carea carea ." ". carea carea carea carea carea carea ." ". . . . . . . .";
    grid-template-rows: 70px 24px 24px 24px repeat(3,1fr) repeat(1,48px);
    grid-template-columns: 10px 1fr repeat(4,3fr) 1fr 10px;
  }
}

<div
  className="c0"
/>
`;

exports[`<Page /> should render styles properly 2`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: carea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0 0 2px 0 rgba(0,0,0,0.12),0 2px 2px 0 rgba(0,0,0,0.24);
}

<div
  className="c0"
/>
`;

exports[`<Page /> should render styles properly 3`] = `
.c0 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  grid-area: carea;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;
